### PR TITLE
Implement float/int conversions in backend 

### LINKED ISSUE #323: Implement BIR instructions for float/int conversion
Issue #321 will introduce new BIR instructions for float/int conversion.

These need to be implemented in the backend using #288.

This issue should wait for #321.

This is part of #262.

### PR DESCRIPTION
**Pull request by manuranga**
Implement float/int conversions in backend
Fixes  #323

### CODE REVIEW

File: compiler/modules/front/codeGen.bal [hunk 1]
```diff
@@ -1113,12 +1113,10 @@ function codeGenTypeCast(CodeGenContext cx, bir:BasicBlock bb, Environment env,
...
     t:UniformTypeBitSet? toNumType = t:singleNumericType(toType);
-    if toNumType != ()
-       // Remove the next line to enable conversion when support in backend is implemented
-       && toNumType == t:NEVER 
-       && !t:isSubtypeSimple(t:intersect(reg.semType, t:NUMBER), toNumType) {
+    if toNumType != () && !t:isSubtypeSimple(t:intersect(reg.semType, t:NUMBER), toNumType) {
         toType = t:diff(toType, t:diff(t:NUMBER, toNumType));
-        bir:Register result = cx.createRegister(t:intersect(reg.semType, toType));
+        
+        bir:Register result = cx.createRegister(t:union(t:intersect(t:complement(t:NUMBER), reg.semType), toType));
```
Comment (jclark) [inline compiler/modules/front/codeGen.bal]: That doesn’t look right. Static type if `<T>E` is intersection of T and static type of E.

Comment (jclark) [reply compiler/modules/front/codeGen.bal]: If you are doing ConvertToInt on something with type T, then the type is int|(T - number)

Comment (jclark) [reply compiler/modules/front/codeGen.bal]: Compare with the type calculation in verify.bal.

Comment (manuranga) [reply compiler/modules/front/codeGen.bal]: > int|(T - number)

This is what I was trying to do, I though `T - number` and `T ∩ ∁(number)` are the same. Let me give another try.

Comment (manuranga) [reply compiler/modules/front/codeGen.bal]: >Compare with the type calculation in verify.bal.

Not sure what you meant by this, but I just added bir verifications for these two instructions.

**Pull request merged**